skopt/gp_opt.py

```diff
@@ -257,9 +281,23 @@ def gp_minimize(func, dimensions, base_estimator=None, acq="EI", xi=0.01,
                         args=(gp, np.min(yi), acq, xi, kappa),
                         bounds=space.transformed_bounds,
                         approx_grad=True, maxiter=10)
-
-                if a < best:
-                    next_x, best = x, a
+                    X.append(x)
+                    values.append(a)
+
+            X = np.asarray(X)
+            values = np.asarray(values)
+
+            if is_discrete_space:
```

Comment:
On a side note, not sure it is a good strategy to use `lbfgs` for these spaces. I sent a PR at https://github.com/scikit-optimize/scikit-optimize/pull/172 to fix that.